Compilation error when trying to use palette

When adding any use statement from my local palette crate  to my library -- I have not tried using the official version -- I get the following error:

```text
error[E0275]: overflow evaluating the requirement `f64: std::ops::Div<palette::blend::PreAlpha<_>>`
  --> src\text.rs:35:50
   |
35 |         let centering_height_shift = self.height / 2.0;
   |                                                  ^
   |
   = help: consider increasing the recursion limit by adding a `#![recursion_limit = "256"]` attribute to your crate (`web_utils`)
   = note: required because of the requirements on the impl of `std::ops::Div<palette::blend::PreAlpha<palette::blend::PreAlpha<_>>>` for `f64`
   = note: 127 redundant requirements hidden
   = note: required because of the requirements on the impl of `std::ops::Div<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<palette::blend::PreAlpha<_>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>` for `f64`
```

The error goes away, when I comment out the [line](https://github.com/Ogeon/palette/blob/d4cae1e2510205f7626e880389e5e18b45913bd4/palette/src/blend/pre_alpha.rs#L326)
`impl_scalar_binop!(Div::div, DivAssign::div_assign, [f32, f64]);`

I believe is unrelated to my local changes and it certainly is unrelated to the code, where the error occurs. It may be related to a [bug ](https://stackoverflow.com/questions/70542062) in the rust compiler.

--- a/palette/Cargo.toml
+++ b/palette/Cargo.toml
@@ -38,6 +38,10 @@ std = ["approx?/std"]
 [lib]
 bench = false
 
+[[example]]
+name = "issue_283"
+path = "regression_tests/issue_283.rs"
+
 [dependencies]
 palette_derive = { version = "0.7.0", path = "../palette_derive" }
 fast-srgb8 = "1.0.0"
@@ -76,6 +80,7 @@ serde = "1"
 serde_derive = "1"
 serde_json = "1"
 enterpolation = "0.2.0"
+scad = "1.2.2" # For regression testing #283
 
 [dev-dependencies.clap]
 version = "3.2.23"

--- a/palette/regression_tests/issue_283.rs
+++ b/palette/regression_tests/issue_283.rs
@@ -1,0 +1,13 @@
+// Checks that issue #283 (fixed in 0.7.1) doesn't re-appear. The cause was the
+// existence of `impl Mul<PreAlpha<C>> for f32` and `impl Mul<PreAlpha<C>> for
+// f64`.
+
+// Both of these uses are necessary for triggering the issue
+#[allow(unused_imports)]
+use palette::Oklch;
+#[allow(unused_imports)]
+use scad::OffsetType;
+
+fn main() {
+    println!("{}", 42.0 * 1.0); // bug also happens when specifying f32 or f64
+}

--- a/palette/src/blend/pre_alpha.rs
+++ b/palette/src/blend/pre_alpha.rs
@@ -296,20 +296,22 @@ macro_rules! impl_scalar_binop {
                 }
             }
 
-            impl<C> $op_trait<PreAlpha<C>> for $ty
-            where
-                C: Premultiply<Scalar = $ty>,
-                $ty: $op_trait<$ty, Output = $ty> + $op_trait<C, Output = C>,
-            {
-                type Output = PreAlpha<C>;
-
-                fn $op_trait_fn(self, color: PreAlpha<C>) -> Self::Output {
-                    PreAlpha {
-                        color: $op_trait::<C>::$op_trait_fn(self, color.color),
-                        alpha: $op_trait::<$ty>::$op_trait_fn(self, color.alpha),
-                    }
-                }
-            }
+            // // Disabled as work-around for https://github.com/Ogeon/palette/issues/283
+            // // Blocked by https://github.com/rust-lang/rust/issues/80542
+            // impl<C> $op_trait<PreAlpha<C>> for $ty
+            // where
+            //     C: Premultiply<Scalar = $ty>,
+            //     $ty: $op_trait<$ty, Output = $ty> + $op_trait<C, Output = C>,
+            // {
+            //     type Output = PreAlpha<C>;
+            //
+            //     fn $op_trait_fn(self, color: PreAlpha<C>) -> Self::Output {
+            //         PreAlpha {
+            //             color: $op_trait::<C>::$op_trait_fn(self, color.color),
+            //             alpha: $op_trait::<$ty>::$op_trait_fn(self, color.alpha),
+            //         }
+            //     }
+            // }
 
             impl<C> $op_assign_trait<$ty> for PreAlpha<C>
             where

